liboptv/src/multimed.c

```diff
@@ -0,0 +1,656 @@
+  
+  
+  nz =  cal->mmlut.nz;
+  nr =  cal->mmlut.nr;
+    
+  /* check whether point is inside camera's object volume */
+  if (ir > nr)              return (0);
+  if (iz < 0  ||  iz > nz)  return (0);
+  
+  /* bilinear interpolation in r/z box */
+  /* ================================= */
+  
+  /* get vertices of box */
+  v4[0] = ir*nz + iz;
+  v4[1] = ir*nz + (iz+1);
+  v4[2] = (ir+1)*nz + iz;
+  v4[3] = (ir+1)*nz + (iz+1);
+  
+  /* 2. check wther point is inside camera's object volume */
+  /* important for epipolar line computation */
+  for (i=0; i<4; i++)
+    if (v4[i] < 0  ||  v4[i] > nr*nz)   return (0);
+  
+  /* interpolate */
+  mmf = cal->mmlut.data[v4[0]] * (1-sr)*(1-sz)
+    + cal->mmlut.data[v4[1]] * (1-sr)*sz
+    + cal->mmlut.data[v4[2]] * sr*(1-sz)
+    + cal->mmlut.data[v4[3]] * sr*sz;
+  
+  return (mmf);
+  
+    
+  }  
+}
+
+
+/*  volumedimension() finds the limits of the measurement volume in 3D space
+    Arguments:
+    double pointers to the limits of the volume in x (xmin,xmax), y (ymin, ymax) and
+    z (zmin, zmax) directions
+    pointer to the volume parameters vpar
+    pointer to the control parameters cpar
+    pointer to the Calibration parameters cal
+*/  
+void volumedimension (double *xmax
+                    , double *xmin
+                    , double *ymax
+                    , double *ymin
+                    , double *zmax
+                    , double *zmin
+                    , volume_par *vpar
+                    , control_par *cpar
+                    , Calibration *cal){
+
+  int   i_cam, i, j;
+  double X, Y, Z, R, Rmax=0, Zmin, Zmax;
+  vec3d pos, a, xyz, xyz_t;
+  double x,y;  
+  double xc[2], yc[2];  /* image corners */
+  Exterior Ex_t;
+  double X_t, Y_t, Z_t, Zmin_t,Zmax_t;
+  double        cross_p[3],cross_c[3];
+  
+  xc[0] = 0.0;
+  xc[1] = (double) cpar->imx;
+  yc[0] = 0.0;
+  yc[1] = (double) cpar->imy;
+  
+  
+  
+       Zmin = vpar->Zmin_lay[0];
+       Zmax = vpar->Zmax_lay[0];
+
+       *zmin = Zmin;
+       *zmax = Zmax;
+ 
+
+    
+  /* find extrema of imaged object volume */
+  /* ==================================== */
+  
+
+  for (i_cam = 0; i_cam < cpar->num_cams; i_cam++){
+      for (i = 0; i < 2; i ++) for (j = 0; j < 2; j++) {
+        
+//        *xmax = vpar->X_lay[1];
+//        *xmin = vpar->X_lay[0];
+ 
+  
+      /* intersect with image vertices rays */
+/*
+          pixel_to_metric (0.0, 0.0, imx,imy, pix_x,pix_y, &x,&y, chfield);
+          x = x - I.xh;
+          y = y - I.yh;
+          correct_brown_affin (x, y, ap, &x,&y);
+          ray_tracing(x,y, Ex, I, G, mmp, &X1, &Y1, &Z1, &a, &b, &c);
+          Z = Zmin;   X = X1 + (Z-Z1) * a/c;   Y = Y1 + (Z-Z1) * b/c;
+          R = sqrt (  (X-Ex.x0)*(X-Ex.x0)
+              + (Y-Ex.y0)*(Y-Ex.y0)); 
+
+*/ 
+          pixel_to_metric (&x, &y, xc[i], yc[j], cpar);
+          x = x - cal->int_par.xh;
+          y = y - cal->int_par.yh;
+          correct_brown_affin (x, y, cal->added_par, &x, &y);

```

Comment:
``cal`` is supposed to be an array, and it should be ``cal[i_cam]``.
The test for volumedimension should be complicated somewhat to make sure the current code fails.